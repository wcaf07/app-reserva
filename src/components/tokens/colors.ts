const colors = {
  blueLight: '#A4C64A',
  greenLight: '#46A2F8',
  white: '#ffffff',
};

export default colors;
