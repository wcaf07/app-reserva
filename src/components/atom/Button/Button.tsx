import * as React from 'react';
import colors from '@tokens/colors';
import styled from 'styled-components/native';

type ButtonProps = {
  text: string;
  color: string;
  onPress: () => void;
  testID?: string;
};

const Container = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 15px;
  background-color: ${(props) => (props.color ? props.color : colors.white)};
`;

const TextButton = styled.Text`
  color: ${colors.white};
  font-weight: bold;
`;

const Button: React.FC<ButtonProps> = (props) => {
  return (
    <Container
      color={props.color}
      onPress={props.onPress}
      testID={props.testID}>
      <TextButton>{props.text}</TextButton>
    </Container>
  );
};

export default Button;
