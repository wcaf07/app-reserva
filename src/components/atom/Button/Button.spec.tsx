import React from 'react';
import {render} from '@testing-library/react-native';
import Button from './Button';
import colors from '@/components/tokens/colors';

describe('Button.styles', () => {
  test('THEN Button SHOULD match with snapshot when is show', () => {
    const component = render(
      <Button text="test" color={colors.blueLight} onPress={() => {}} />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
