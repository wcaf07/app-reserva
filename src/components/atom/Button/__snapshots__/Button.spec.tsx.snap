// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.styles THEN Button SHOULD match with snapshot when is show 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#46A2F8",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 15,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 15,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontWeight": "bold",
        },
      ]
    }
  >
    test
  </Text>
</View>
`;
